app/src/main/java/de/grobox/transportr/trips/search/DirectionsActivity.java

```diff
@@ -147,6 +152,26 @@ private void processIntent() {
 					to = viewModel.getWork().getValue();
 					if (to == null) new WorkPickerFragment().show(getSupportFragmentManager(), WorkPickerFragment.TAG);
 					break;
+				case INTENT_URI_FAVORITE:
+					from = null;
+					via = null;
+					to = null;
+					long uid = intent.getLongExtra("uid",0);
+
+					LiveData<List<FavoriteTripItem>> tripsLiveData;
+					tripsLiveData = viewModel.getFavoriteTrips();
+					tripsLiveData.observe(this, trips -> {
```

Comment:
You could just do `viewModel.getFavoriteTrips().observe(...)`

Reply:
Yes, but I use `tripsLiveData` to remove the observer when the actual trip is found.
Otherwise the found trip will be found again on every update of the livedata with the same trips.

Reply:
Ah right. You need a reference to the LiveData.